llvm/lib/Target/SystemZ/SystemZFrameLowering.cpp

```diff
@@ -1449,7 +1449,12 @@ void SystemZXPLINKFrameLowering::inlineStackProbe(
   fullyRecomputeLiveIns({StackExtMBB, NextMBB});
 }
 
-bool SystemZXPLINKFrameLowering::hasFP(const MachineFunction &MF) const {
+bool SystemZXPLINKFrameLowering::hasFPImpl(const MachineFunction &MF) const {
+  // Naked functions have no stack frame pushed, so we don't have a frame
+  // pointer.
+  if (MF.getFunction().hasFnAttribute(Attribute::Naked))
+    return false;
+
```

Comment:
That is already checked in the parent class, isn't it?

Reply:
Ah, yeah, looks like a leftover from an earlier version of the patch. I guess I missed it because there are two implementations of `hasFPImpl()` in this file.